sklbench/benchs/sklearn_estimator.py

```diff
@@ -0,0 +1,405 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import io
+import json
+import argparse
+import logging
+from importlib.metadata import version, PackageNotFoundError
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+from sklearn.metrics import (
+    # accuracy metrics
+    accuracy_score, balanced_accuracy_score, roc_auc_score, log_loss,
+    # regression metrics
+    mean_squared_error, r2_score,
+    # clustering metrics
+    davies_bouldin_score, homogeneity_score, completeness_score)
+from typing import Dict, List, Union
+from ..utils.custom_types import BenchCase, Numeric, NumpyNumeric
+from ..utils.logger import logger
+from ..utils.config import bench_case_filter
+from ..utils.common import (
+    custom_format, get_module_members,
+    get_bench_case_value, get_data_name)
+from ..utils.measurement import measure_case
+from ..datasets import load_data
+from ..utils.data import split_and_transform_data
+from ..utils.special_params import assign_case_special_values_on_run
+
+
+def get_estimator(library_name: str, estimator_name: str):
+    classes_map, _ = get_module_members(library_name.split('.'))
+    if estimator_name not in classes_map:
+        raise ValueError(
+            f'Unable to find {estimator_name} estimator in {library_name} module.')
+    if len(classes_map[estimator_name]) != 1:
+        logger.debug(f'List of estimator with name "{estimator_name}": '
+                     f'{classes_map[estimator_name]}')
+        logger.warning(f'Found {len(classes_map[estimator_name])} classes for '
+                       f'"{estimator_name}" estimator name. '
+                       f'Using first {classes_map[estimator_name][0]}.')
+    estimator = classes_map[estimator_name][0]
+    if not issubclass(estimator, BaseEstimator):
+        logger.info(f'{estimator} estimator is not derived from sklearn\'s BaseEstimator')
+    return estimator
+
+
+def get_estimator_methods(bench_case: BenchCase) -> Dict[str, List[str]]:
+    # default estimator methods
+    estimator_methods = {
+        'training': ['fit'],
+        'inference': ['predict', 'predict_proba', 'transform']
+    }
+    for stage in estimator_methods.keys():
+        methods = get_bench_case_value(
+            bench_case, f'algorithm:estimator_methods:{stage}', None)
+        if methods is not None:
+            estimator_methods[stage] = methods.split('|')
+    return estimator_methods
+
+
+def estimator_to_task(estimator_name: str) -> str:
+    estimator_groups = {

```

Comment:
It feels like to me that this is the wrong location to have these groups stored. If any additional algos get added, then someone should find somewhere near the configs to modify (or at least follow standards observed elsewhere in the refactor)

Reply:
Separate JSON file with mapping was added near `benchmarks/sklearn_estimator.py`. Not near to configs because their dir is expected to have only benchmarking spec-compatible files and root of repo is not suitable for this too.